platform/posix/transport/src/sockets_posix.c

```diff
@@ -0,0 +1,438 @@
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/* Standard includes. */
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* POSIX socket includes. */
+#include <errno.h>
+#include <netdb.h>
+#include <time.h>
+#include <poll.h>
+#include <unistd.h>
+#include <arpa/inet.h>
+
+#include <sys/socket.h>
+#include <sys/types.h>
+
+#include "sockets_posix.h"
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Number of DNS records to attempt a connection.
+ *
+ * @note Negative value implies attempting to connect to all DNS records
+ * until successful.
+ */
+#define NUM_DNS_RECORDS_TO_TRY    ( -1 )
+
+/**
+ * @brief Number of milliseconds in one second.
+ */
+#define ONE_SEC_TO_MS             ( 1000 )
+
+/**
+ * @brief Number of microseconds in one millisecond.
+ */
+#define ONE_MS_TO_US              ( 1000 )
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Defined by transport layer to check send or receive error.
+ */
+extern int errno;
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Resolve a host name.
+ *
+ * @brief param[in] pHostName Server host name.
+ * @brief param[in] hostNameLength Length associated with host name.
+ * @brief param[out] pListHead List containing resolved DNS records.
+ *
+ * @return #SOCKETS_SUCCESS if successful; #SOCKETS_DNS_FAILURE, #SOCKETS_CONNECT_FAILURE on error.
+ */
+static SocketStatus_t resolveHostName( const char * pHostName,
+                                       size_t hostNameLength,
+                                       struct addrinfo ** pListHead );
+
+/**
+ * @brief Traverse list of DNS records and return successfully if able to connect to one.
+ *
+ * @brief param[in] pListHead List containing resolved DNS records.
+ * @brief param[in] pHostName Server host name.
+ * @brief param[in] hostNameLength Length associated with host name.
+ * @brief param[in] port Server port in host-order.
+ * @brief param[in] pTcpSocket Pointer to the socket descriptor.
```

Comment:
This is output parameter.

```
@param[out] pTcpSocket The output parameter to return the created socket.
```

Also, @brief before params is incorrect. It should be @param as above. Please correct at all places.